harness/determined/launch/torch_distributed.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import os
+import subprocess
+import sys
+from typing import List
+
+import determined as det
+
+
+def main(script: List[str]) -> int:
+    info = det.get_cluster_info()
+    os.environ["USE_TORCH_DIST"] = "True"
+
+    chief_ip = info.container_addrs[0]
+    os.environ["DET_CHIEF_IP"] = chief_ip
+
+    torch_distributed_cmd = [
+        "python3",
+        "-m",
+        "torch.distributed.run",
+        "--nnodes",
+        str(len(info.container_addrs)),
+        "--nproc_per_node",
+        str(len(info.slot_ids)),
+        "--node_rank",
+        str(info.container_rank),
+        "--max_restarts",
+        "1",
+        "--master_addr",
+        "localhost" if len(info.container_addrs) == 1 else chief_ip,
+        "--master_port",
+        "29500",
+        "--module"
+    ]
+
+    log_redirect_cmd = [
+        "determined.exec.worker_process_wrapper",
+        "RANK",
+        "--",
+    ]
+
+    print(f"launching {torch_distributed_cmd + log_redirect_cmd + script}")
+
+    return subprocess.Popen(
+        torch_distributed_cmd + log_redirect_cmd + script
+    ).wait()
+
+
+def parse_args(args: List[str]) -> List[str]:
+    parser = argparse.ArgumentParser(
+        usage="%(prog)s (--trial TRIAL)|(SCRIPT...)",
```

Comment:
do we want to allow overrides to the `torch.distributed.run` command, the way we do for horovodrun?

We need to decide now because it is a breaking change to add them later.

Reply:
added. it should be supported since quite a few parameters like rendezvous backend/port should be able to be overriden.